godot-core/src/builtin/string/gstring.rs

```diff
@@ -148,6 +148,16 @@ impl GString {
         *boxed
     }
 
+    /// Convert a `GString` sys pointer to a mutable reference with unbounded lifetime.
+    ///
+    /// # Safety
+    ///
+    /// `ptr` must point to a live `GString` for the duration of `'a`.
+    pub(crate) unsafe fn borrow_string_sys_mut<'a>(ptr: sys::GDExtensionStringPtr) -> &'a mut Self {
```

Comment:
There's another safety precondition: no other Rust reference to that location must exist, otherwise `&mut` is not exclusive -> immediate UB.

(also, you can start directly after `# Safety` on the next line, without empty line, we're not very consistent with it yet)